ras_party/support/session_decorator.py

```diff
@@ -1,15 +1,35 @@
-from functools import wraps
 import logging
+from functools import wraps
 
-from flask import current_app
 import structlog
+from flask import current_app
 
-from ras_party.exceptions import RasError, RasDatabaseError
-
+from ras_party.exceptions import RasError
 
 logger = structlog.wrap_logger(logging.getLogger(__name__))
 
 
+def handle_session(f, args, kwargs):
+    logger.info("Acquiring database session.")
+    session = current_app.db.session()
+    try:
+        result = f(*args, session=session, **kwargs)
+        logger.info("Committing database session.")
+        session.commit()
+        return result
+    except RasError:
+        logger.error(f"Rolling back database session due to failure executing function", function=f.__name__)
```

Comment:
I don't see the point of logging at error here then exception in the error handler.

Reply:
Fair point the function info will be in the traceback

Reply:
Actually rethinking. It does have the context of rolling back the session so I think it's worth being here

Reply:
This should probably be handled at the source of the exception. I was thinking some more any maybe we should just get rid of this session_decorator.py and create transactions and handle them using http://flask.pocoo.org/docs/0.12/api/#flask.Flask.before_request and http://flask.pocoo.org/docs/0.12/api/#flask.Flask.after_request